sdk/core/System.ClientModel/src/Convenience/AsyncEnumerableResultOfT.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel;
+
+#pragma warning disable CS1591 // public XML comments
+public abstract class AsyncEnumerableResult<T> : ClientResult, IAsyncEnumerable<T>, IAsyncDisposable
+    where T : IPersistableModel<T>
+{
+    protected internal AsyncEnumerableResult(PipelineResponse response) : base(response)
```

Comment:
can the response be updated/changed?

Reply:
I don't think it can.  Is there a scenario where we would want that?
(Not sure I understand the implications of your question.)

Reply:
The scenario is using this type to expose pageable REST APIs. In such scenarios there are multiple responses representing all items in the collection/pageable

Reply:
Would you want to do that by changing the value returned from the response?  Why not return a collection of PipelineResponses instead? (i.e. not inherit from ClientResult)

Reply:
Addressed.  I see that we already do that in Azure.Core Operation implementations such as [KeyVaultBackupOperation](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultBackupOperation.cs#L138-L142) and [FuzzySearchBatchOperation](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/maps/Azure.Maps.Search/src/Operations/FuzzySearchBatchOperation.cs#L180-L182).